src/libraries/System.Runtime/tests/System.Runtime.Extensions.Tests/System/EnvironmentTests.cs

```diff
@@ -574,6 +574,45 @@ public void GetLogicalDrives_Windows_MatchesExpectedLetters()
             }
         }
 
+        [Fact]
+        public void TestCpuUsage()
+        {
+            if ((OperatingSystem.IsIOS() && !OperatingSystem.IsMacCatalyst()) || PlatformDetection.IstvOS || PlatformDetection.IsBrowser)
+            {
+                // Environment should return 0 for all values
+                Environment.ProcessCpuUsage usage = Environment.CpuUsage;
+                Assert.Equal(TimeSpan.Zero, usage.UserTime);
+                Assert.Equal(TimeSpan.Zero, usage.PrivilegedTime);
+                Assert.Equal(TimeSpan.Zero, usage.TotalTime);
+            }
+            else
+            {
+                Process currentProcess = Process.GetCurrentProcess();
+
+                TimeSpan userTime = currentProcess.UserProcessorTime;
+                TimeSpan privilegedTime = currentProcess.PrivilegedProcessorTime;
+                TimeSpan totalTime = currentProcess.TotalProcessorTime;
+
+                Environment.ProcessCpuUsage usage = Environment.CpuUsage;
+                Assert.True(usage.UserTime.TotalMilliseconds >= 0);
+                Assert.True(usage.PrivilegedTime.TotalMilliseconds >= 0);
+                Assert.True(usage.TotalTime.TotalMilliseconds >= 0);
+                Assert.Equal(usage.TotalTime, usage.UserTime + usage.PrivilegedTime);
+
+                Assert.True(usage.UserTime >= userTime);
+                Assert.True(usage.PrivilegedTime >= privilegedTime);
+                Assert.True(usage.TotalTime >= totalTime);
+
+                TimeSpan delta = TimeSpan.FromMinutes(1);
+
+                Assert.True(usage.UserTime - userTime < delta);
+                Assert.True(usage.PrivilegedTime - privilegedTime < delta);
+                Assert.True(usage.TotalTime - totalTime < delta);
+
+                Thread.Sleep(100);
+            }
```

Comment:
Would any of these tests fail if we had a bug where, for example, we reported 5ms instead as 5us?

Reply:
I am not sure we can be that accurate without writing the tests calling the underlying OS. I tried to look at the Process tests and couldn't find a test doing that either.